src/vs/platform/quickinput/browser/quickInput.ts

```diff
@@ -1075,7 +1075,9 @@ class QuickPick<T extends IQuickPickItem> extends QuickInput implements IQuickPi
 			if (this.title) {
 				ariaLabel += ` - ${this.title}`;
 			}
+			this.ui.inputBox.setAttribute('aria-label', ariaLabel!);
```

Comment:
this change, and the change in `quickInputBox.ts` isn't needed because the placeholder will be read by the screen reader, right? You only need the line in the `gettingStarted.ts`

Reply:
The placeholder isn't always read by every screen reader. Please see the "Placeholder text" section in https://www.w3.org/WAI/tutorials/forms/instructions/ 
<quote>
placeholder text is not a replacement for labels. Assistive technologies, such as screen readers, do not treat placeholder text as labels. Moreover, at the time of writing this tutorial, placeholder text is not broadly supported across assistive technologies and not displayed in older web browsers.
</quote>

Reply:
Nice find. Can you instead implement this by following how we set the placeholder:
https://github.com/microsoft/vscode/blob/a02490de3c0fe0d108685dcfe60daf79c8e58062/src/vs/platform/quickinput/browser/quickInputBox.ts#L74-L80

have this call into the `QuickInputBox` to set the `ariaLabel` and then have that set the aria label via:
```ts
this.findInput.inputBox.setAriaLabel(...)
```

follow the existing handling of `placeholder` to make sure you handle all the cases.

Reply:
added a setAriaLabel in quickInputBox.js and used it. I am not sure why you need a get/set arialabel in quickInputBox.js. I am happy to add these if needed.